requests/utils.py

```diff
@@ -732,6 +732,22 @@ def to_native_string(string, encoding='ascii'):
 
     return out
 
+def validate_header(header):
+    """Verifies that header value doesn't contain leading whitespace or
+    return characters. This prevents unintended header injection.
+
+    header is a tuple, in the format (name, value)
+    """
+    name, value = header
+
+    if isinstance(value, bytes):
+        pat = re.compile(b'^\S[^\\r\\n]*$|^$')
+    else:
+        pat = re.compile(r'^\S[^\r\n]*$|^$')
```

Comment:
Or even just hoist them directly above this function. Let's prefix their names with underscores just to discourage people relying on them.